plugins/ptp_operator/metrics/metrics.go

```diff
@@ -225,149 +308,288 @@ func (p *PTPEventManager) MockTest(t bool) {
 							}
 						}
-					})
-				}(p, threshold, iface)
+
+					}(p, threshold.HoldOverTimeout, iface, lastRole, threshold.Close)
+				}
 			}
 		}
 	}
 }
 
 // GenPhc2SysEvent ... generate events form the logs
-func (p *PTPEventManager) GenPhc2SysEvent(iface string, offsetFromMaster int64, clockState ceevent.SyncState) {
+func (p *PTPEventManager) GenPhc2SysEvent(ifaces []types.IFace, offsetFromMaster int64, clockState ceevent.SyncState) {
 	defer func() {
 		if r := recover(); r != nil {
 			fmt.Println("Recovered in f", r)
 		}
 	}()
-	lastClockState := p.Stats[iface].lastClockState
-	threshold := p.PtpThreshold(iface)
+	if clockState == "" {
+		return
+	}
+	iface := ifaces[0] // use first interface and copy to all
+	lastClockState := p.Stats[iface].lastSyncState
+	threshold := p.PtpThreshold(string(iface))
 	switch clockState {
 	case ceevent.LOCKED:
 		switch lastClockState {
 		case ceevent.FREERUN: //last state was already sent for FreeRUN , but if its within then send again with new state
 			if offsetFromMaster < threshold.MaxOffsetThreshold && offsetFromMaster > threshold.MinOffsetThreshold { // within range
-				p.PublishEvent(clockState, offsetFromMaster, iface, channel.PTPEvent) // change to locked
-				p.Stats[iface].lastClockState = clockState
-				p.Stats[iface].lastOffset = int64(offsetFromMaster)
-				p.Stats[iface].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				p.PublishEvent(clockState, offsetFromMaster, string(iface), channel.PTPEvent) // change to locked
+				for _, i := range ifaces {
+					p.Stats[i].lastSyncState = clockState
+					p.Stats[i].lastOffset = int64(offsetFromMaster)
+					p.Stats[i].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				}
+
 			}
 		case ceevent.LOCKED: // last state was in sync , check if it is out of sync now
 			if offsetFromMaster > threshold.MaxOffsetThreshold || offsetFromMaster < threshold.MinOffsetThreshold { // out of sync
-				p.PublishEvent(ceevent.FREERUN, offsetFromMaster, iface, channel.PTPEvent)
-				p.Stats[iface].lastClockState = ceevent.FREERUN
-				p.Stats[iface].lastOffset = int64(offsetFromMaster)
-				p.Stats[iface].reset()
+				p.PublishEvent(ceevent.FREERUN, offsetFromMaster, string(iface), channel.PTPEvent)
+				for _, i := range ifaces {
+					p.Stats[i].lastSyncState = ceevent.FREERUN
+					p.Stats[i].lastOffset = int64(offsetFromMaster)
+				}
 			} else {
-				p.Stats[iface].lastOffset = int64(offsetFromMaster)
-				p.Stats[iface].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				for _, i := range ifaces {
+					p.Stats[i].lastOffset = int64(offsetFromMaster)
+					p.Stats[i].addValue(int64(offsetFromMaster)) // update off set when its in locked state and hold over only
+				}
 			}
 		case ceevent.HOLDOVER: // last state was holdover, since it is in locked now check if it qualifies to be locked
```

Comment:
the comment followed can be deleted since it is contradict with the comment in the next line.

Reply:
done